cinema/permissions.py

```diff
@@ -0,0 +1,13 @@
+from rest_framework.permissions import BasePermission, SAFE_METHODS
+
+
+class IsAdminOrIfAuthenticatedOrReadOnly(BasePermission):
+
+    def has_permission(self, request, view):
+        return bool(
+            request.method in SAFE_METHODS and
+            request.user and
+            request.user.is_authenticated
```

Comment:
The logic here seems to be incorrect for the intended permission class. The condition `request.method in SAFE_METHODS and request.user and request.user.is_authenticated` will only allow safe methods if the user is authenticated. If the intention is to allow safe methods for any user, authenticated or not, you should remove `request.user and request.user.is_authenticated` from this condition.